File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task156.py

class TreeNode:
    def __init__(self, val):
        if not isinstance(val, int):
            raise TypeError("Node value must be an integer")
        self.val = val
        self.left = None
        self.right = None

def preorder_traversal(root):
    result = []
    if root is None:
        return result
    
    def preorder_helper(node):
        if node is None:
            return
        result.append(node.val)
        preorder_helper(node.left)
        preorder_helper(node.right)
    
    preorder_helper(root)
    return result

def inorder_traversal(root):
    result = []
    if root is None:
        return result
    
    def inorder_helper(node):
        if node is None:
            return
        inorder_helper(node.left)
        result.append(node.val)
        inorder_helper(node.right)
    
    inorder_helper(root)
    return result

def postorder_traversal(root):
    result = []
    if root is None:
        return result
    
    def postorder_helper(node):
        if node is None:
            return
        postorder_helper(node.left)
        postorder_helper(node.right)
        result.append(node.val)
    
    postorder_helper(root)
    return result

if __name__ == "__main__":
    # Test Case 1: Single node tree
    test1 = TreeNode(1)
    print("Test 1 - Preorder:", preorder_traversal(test1))
    print("Test 1 - Inorder:", inorder_traversal(test1))
    print("Test 1 - Postorder:", postorder_traversal(test1))
    
    # Test Case 2: Balanced tree
    test2 = TreeNode(1)
    test2.left = TreeNode(2)
    test2.right = TreeNode(3)
    test2.left.left = TreeNode(4)
    test2.left.right = TreeNode(5)
    print("\\nTest 2 - Preorder:", preorder_traversal(test2))
    print("Test 2 - Inorder:", inorder_traversal(test2))
    print("Test 2 - Postorder:", postorder_traversal(test2))
    
    # Test Case 3: Left-skewed tree
    test3 = TreeNode(1)
    test3.left = TreeNode(2)
    test3.left.left = TreeNode(3)
    print("\\nTest 3 - Preorder:", preorder_traversal(test3))
    print("Test 3 - Inorder:", inorder_traversal(test3))
    print("Test 3 - Postorder:", postorder_traversal(test3))
    
    # Test Case 4: Right-skewed tree
    test4 = TreeNode(1)
    test4.right = TreeNode(2)
    test4.right.right = TreeNode(3)
    print("\\nTest 4 - Preorder:", preorder_traversal(test4))
    print("Test 4 - Inorder:", inorder_traversal(test4))
    print("Test 4 - Postorder:", postorder_traversal(test4))
    
    # Test Case 5: Empty tree
    test5 = None
    print("\\nTest 5 - Preorder:", preorder_traversal(test5))
    print("Test 5 - Inorder:", inorder_traversal(test5))
    print("Test 5 - Postorder:", postorder_traversal(test5))

File: dataset/CoT/CLAUDE_claude-sonnet-4-5-20250929/Task156.java

import java.util.*;

class Task156 {
    static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        
        TreeNode(int val) {
            if (val < Integer.MIN_VALUE || val > Integer.MAX_VALUE) {
                throw new IllegalArgumentException("Invalid node value");
            }
            this.val = val;
            this.left = null;
            this.right = null;
        }
    }
    
    public static List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        if (root == null) {
            return result;
        }
        preorderHelper(root, result);
        return result;
    }
    
    private static void preorderHelper(TreeNode node, List<Integer> result) {
        if (node == null) {
            return;
        }
        result.add(node.val);
        preorderHelper(node.left, result);
        preorderHelper(node.right, result);
    }
    
    public static List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        if (root == null) {
            return result;
        }
        inorderHelper(root, result);
        return result;
    }
    
    private static void inorderHelper(TreeNode node, List<Integer> result) {
        if (node == null) {
            return;
        }
        inorderHelper(node.left, result);
        result.add(node.val);
        inorderHelper(node.right, result);
    }
    
    public static List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> result = new ArrayList<>();
        if (root == null) {
            return result;
        }
        postorderHelper(root, result);
        return result;
    }
    
    private static void postorderHelper(TreeNode node, List<Integer> result) {
        if (node == null) {
            return;
        }
        postorderHelper(node.left, result);
        postorderHelper(node.right, result);
        result.add(node.val);
    }
    
    public static void main(String[] args) {
        // Test Case 1: Single node tree
        TreeNode test1 = new TreeNode(1);
        System.out.println("Test 1 - Preorder: " + preorderTraversal(test1));
        System.out.println("Test 1 - Inorder: " + inorderTraversal(test1));
        System.out.println("Test 1 - Postorder: " + postorderTraversal(test1));
        
        // Test Case 2: Balanced tree
        TreeNode test2 = new TreeNode(1);
        test2.left = new TreeNode(2);
        test2.right = new TreeNode(3);
        test2.left.left = new TreeNode(4);
        test2.left.right = new TreeNode(5);
        System.out.println("\\nTest 2 - Preorder: " + preorderTraversal(test2));
        System.out.println("Test 2 - Inorder: " + inorderTraversal(test2));
        System.out.println("Test 2 - Postorder: " + postorderTraversal(test2));
        
        // Test Case 3: Left-skewed tree
        TreeNode test3 = new TreeNode(1);
        test3.left = new TreeNode(2);
        test3.left.left = new TreeNode(3);
        System.out.println("\\nTest 3 - Preorder: " + preorderTraversal(test3));
        System.out.println("Test 3 - Inorder: " + inorderTraversal(test3));
        System.out.println("Test 3 - Postorder: " + postorderTraversal(test3));
        
        // Test Case 4: Right-skewed tree
        TreeNode test4 = new TreeNode(1);
        test4.right = new TreeNode(2);
        test4.right.right = new TreeNode(3);
        System.out.println("\\nTest 4 - Preorder: " + preorderTraversal(test4));
        System.out.println("Test 4 - Inorder: " + inorderTraversal(test4));
        System.out.println("Test 4 - Postorder: " + postorderTraversal(test4));
        
        // Test Case 5: Empty tree
        TreeNode test5 = null;
        System.out.println("\\nTest 5 - Preorder: " + preorderTraversal(test5));
        System.out.println("Test 5 - Inorder: " + inorderTraversal(test5));
        System.out.println("Test 5 - Postorder: " + postorderTraversal(test5));
    }
}
